### PR TITLE
LPS-145365 Add tests for guest account creation

### PR DESCRIPTION
**Pull request by pei-jung**

### CODE REVIEW

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/UserAccountResourceTest.java [hunk 1]
```diff
@@ -594,6 +613,48 @@ public void testPostUserAccount() throws Exception {
...
+				LocaleUtil.getDefault(), "Guest"
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		AtomicBoolean called = new AtomicBoolean();
+
+		_testPostUserAccountAsGuest(
+			false, new MockCaptcha(() -> called.set(true)));
+
+		Assert.assertFalse(called.get());
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/UserAccountResourceTest.java:632]: Would it work to just call `Assert.fail()` in the captcha impl?

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:forward

**Pull request closed without merging**